hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/CreateHoodieTableCommand.scala

```diff
@@ -62,12 +65,22 @@ case class CreateHoodieTableCommand(table: CatalogTable, ignoreIfExists: Boolean
     // check if there are conflict between table configs defined in hoodie table and properties defined in catalog.
     CreateHoodieTableCommand.validateTblProperties(hoodieCatalogTable)
 
-    // init hoodie table
-    hoodieCatalogTable.initHoodieTable()
+    val queryAsProp = hoodieCatalogTable.catalogProperties.get(ConfigUtils.IS_QUERY_AS_RO_TABLE)
+    if (queryAsProp.isEmpty) {
```

Comment:
create the first table with `hoodie.query.as.ro.table` param may be failed here?

Reply:
it should be. it must have a normal mor table when you want to create a ro/rt one related to.

Reply:
If the table not exsit, queryAsProp may not need to check. I want to create a not exsit mor table with `hoodie.query.as.ro.table=false`, I think need to consider this scene.

Reply:
in a normal process, spark-hudi create and write a mor table, then create ro/rt tables. 
it's easy to support the scene you mentioned above. but i think a mor table (a real table in filesystem, not necessary in metastore) is a prerequisite.
cc @XuQianJin-Stars WDYT?